modin/experimental/engines/omnisci_on_ray/test/utils.py

```diff
@@ -0,0 +1,173 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import modin.pandas as pd
+import pandas
+import pyarrow as pa
+
+from modin.pandas.test.utils import (
+    df_equals,
+    io_ops_bad_exc,
+    eval_io as general_eval_io,
+)
+
+
+def eval_io(
+    fn_name,
+    comparator=df_equals,
+    cast_to_str=False,
+    check_exception_type=True,
+    raising_exceptions=io_ops_bad_exc,
+    check_kwargs_callable=True,
+    modin_warning=None,
+    md_extra_kwargs=None,
+    *args,
+    **kwargs,
+):
+    """
+    Evaluate I/O operation and do equality check after importing Modin's data to OmniSci.
+
+    Notes
+    -----
+    For parameters description please refer to ``modin.pandas.test.utils.eval_io``.
+    """
+
+    def omnisci_comparator(df1, df2):
+        """Evaluate equality comparison of the passed frames after importing the Modin's one to OmniSci."""
+        if isinstance(df1, (pd.DataFrame, pd.Series)):
+            trigger_import(df1)
+        if isinstance(df2, (pd.DataFrame, pd.Series)):
+            trigger_import(df2)
+        return comparator(df1, df2)
+
+    general_eval_io(
+        fn_name,
+        comparator=omnisci_comparator,
+        cast_to_str=cast_to_str,
+        check_exception_type=check_exception_type,
+        raising_exceptions=raising_exceptions,
+        check_kwargs_callable=check_kwargs_callable,
+        modin_warning=modin_warning,
+        md_extra_kwargs=md_extra_kwargs,
+        *args,
+        **kwargs,
+    )
+
+
+def trigger_import(*dfs):
```

Comment:
this is almost a full copy of the [`asv_bench.benchmarks.utils.common.trigger_import`](https://github.com/modin-project/modin/blob/0a642753af2cd2332ac71166bcddc89b0cace9d5/asv_bench/benchmarks/utils/common.py#L443-L468) function. We can't import that function from `asv_bench`, since ASV stuff is not a part of the modin package, we can't import this function to `asv_bench` from here either, because this will break ASV compatibility with the previous version of modin.